magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/nsb_to_h5.py

```diff
@@ -0,0 +1,89 @@
+"""
+Bash scripts to run LSTnsb.py on all the LST runs by using parallel jobs
+
+Usage: python nsb_level.py (-c config.yaml)
+"""
+
+import glob
+import logging
+
+import numpy as np
+import pandas as pd
+
+__all__ = ["collect_nsb"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def collect_nsb(df_LST):
+    """
+    Here we split the LST runs in NSB-wise .txt files
+
+    Parameters
+    ----------
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+
+    Returns
+    -------
+    :class:`pandas.DataFrame`
+        Same dataframe as the input one, but with NSB values added in the 'nsb' column (for the runs processed by nsb_level.py)
+    """
+    nsb_files = glob.glob("nsb_LST_*.txt")
+    for file_nsb in nsb_files:
+        run = file_nsb.split("_")[3]
+        run = run.split(".")[0]
+        nsb = np.nan
+        with open(file_nsb) as ff:
+            line_str = ff.readline().rstrip("\n")
+            nsb = line_str.split(",")[2]
+        df_LST = df_LST.set_index("LST1_run")
+        df_LST.loc[run, "nsb"] = float(nsb)
+        df_LST = df_LST.reset_index()
```

Comment:
do you need to reset index after each line that is read? this can be pretty slow if the database grown a lot